### PR TITLE
core/priority: add scores to output

### PR DESCRIPTION
**Pull request by corverroos**
Adds explicit scores to resulting priorities. This solves the tie-breaker (if scores are identical) problem by allowing the user of the priority protocol to decide how to handle ties based on their use-case. 

- This simplifies the protocol, no fancy pseudo random shuffling.
- Users can however still decide to implement fancy pseudo random shuffling themselves if scores are identical
- Or users can use the priority result ordering defined by peers with lowest peer IDs.

category: feature
ticket: #798

### CODE REVIEW

File: .pre-commit/run_buf.sh [hunk 1]
```diff
@@ -7,4 +7,4 @@ fi
 
 buf generate
 buf lint
-buf breaking --against '.git#branch=origin/main'
```
Comment (corverroos) [inline .pre-commit/run_buf.sh:10]: disabled this since these are breaking changes, but the protos are still in draft, so this breaking changes are ok.

Comment (dB2510) [reply .pre-commit/run_buf.sh:10]: don't forget to uncomment this later

File: core/corepb/v1/priority.proto [hunk 2]
```diff
@@ -23,8 +23,22 @@ message PriorityMsg {
   bytes signature = 9;
 }
 
-// PriorityTopic defines a single peers priorities for a single topic in the Prioritiser protocol.
-message PriorityTopic {
+// PriorityTopic defines a single peers proposed priorities for a single topic in the Prioritiser protocol.
```
Comment (dB2510) [inline core/corepb/v1/priority.proto:26]:
```suggestion
// PriorityTopicProposal defines a single peers proposed priorities for a single topic in the Prioritiser protocol.
```

File: core/corepb/v1/priority.proto [hunk 3]
```diff
@@ -23,8 +23,22 @@ message PriorityMsg {
...
+message PriorityTopicProposal {
   string topic = 1;
   repeated string priorities = 2;
 }
+
+// PriorityTopic defines the cluster wide resulting priorities for a
```
Comment (dB2510) [inline core/corepb/v1/priority.proto:32]:
```suggestion
// PriorityTopicResult defines the cluster wide resulting priorities for a
```

**Review by dB2510 — APPROVED** [review]

**Review by xenowits — APPROVED** [review]
LGTM!

**Pull request merged**